runtime/tests/checker/attachments_test.go

```diff
@@ -4022,7 +4022,7 @@ func TestCheckAttachmentBaseNonMember(t *testing.T) {
 			access(all) attachment A for R {
 				access(all) let bases: [&R]
 				init() {
-					self.bases = []
+					self.bases = [] as [&R]
```

Comment:
How difficult would it be to allow assignments to `self` without an explicit type annotation?

Reply:
Should be possible I think, have not tried yet. I think we'd have to analyze the LHS of the assignment and see if the expression is a member expression accessing `self`

Reply:
Improved in [53e7cc1](https://github.com/onflow/cadence/pull/3023/commits/53e7cc146de36d6b0c3aa5616a952c8ea7763126)